django/db/backends/sqlite3/schema.py

```diff
@@ -227,6 +227,7 @@ def is_self_referential(f):
         meta = type("Meta", (), meta_contents)
         body_copy["Meta"] = meta
         body_copy["__module__"] = model.__module__
+        body_copy["create_pk_if_needed"] = False
```

Comment:
This is an extremely hacky solution. I have no advice, but adding a new backend-specific attribute and passing it in multiple internal function calls is not something that we should introduce in the code.

Reply:
@felixxm thanks for the review. I updated the PR to remove the primary key field if it gets added - is that a better direction?